subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1526,74 +1527,154 @@ static int cmd_wifi_twt_setup_quick(const struct shell *sh, size_t argc,
 	return 0;
 }
 
-static int cmd_wifi_twt_setup(const struct shell *sh, size_t argc,
-			      char *argv[])
+static int twt_args_to_params(const struct shell *sh, size_t argc, char *argv[],
+				 struct wifi_twt_params *params)
 {
-	struct net_if *iface = net_if_get_first_wifi();
-	struct wifi_twt_params params = { 0 };
-	int idx = 1;
+	int opt;
+	int opt_index = 0;
+	struct getopt_state *state;
 	long value;
+	static const struct option long_options[] = {
+		{"negotiation-type", required_argument, 0, 'n'},
+		{"setup-cmd", required_argument, 0, 'c'},
+		{"dialog-token", required_argument, 0, 't'},
+		{"flow-id", required_argument, 0, 'f'},
+		{"responder", required_argument, 0, 'r'},
+		{"trigger", required_argument, 0, 'T'},
+		{"implicit", required_argument, 0, 'I'},
+		{"announce", required_argument, 0, 'a'},
+		{"wake-interval", required_argument, 0, 'w'},
+		{"interval", required_argument, 0, 'i'},
+		{"wake-ahead-duration", required_argument, 0, 'D'},
+		{"info-disable", required_argument, 0, 'd'},
+		{"exponent", required_argument, 0, 'e'},
+		{"help", no_argument, 0, 'h'},
+		{0, 0, 0, 0}};
 
-	context.sh = sh;
+	params->operation = WIFI_TWT_SETUP;
 
-	params.operation = WIFI_TWT_SETUP;
+	while ((opt = getopt_long(argc, argv, "n:c:t:f:r:T:I:a:t:w:i:D:d:e:h",
+				  long_options, &opt_index)) != -1) {
+		state = getopt_state_get();
+		switch (opt) {
+		case 'n':
+			if (!parse_number(sh, &value, state->optarg, NULL,
+			    WIFI_TWT_INDIVIDUAL, WIFI_TWT_WAKE_TBTT)) {
+				return -EINVAL;
+			}
+			params->negotiation_type = (enum wifi_twt_negotiation_type)value;
+			break;
 
-	if (!parse_number(sh, &value, argv[idx++], NULL, WIFI_TWT_INDIVIDUAL,
-			  WIFI_TWT_WAKE_TBTT)) {
-		return -EINVAL;
-	}
-	params.negotiation_type = (enum wifi_twt_negotiation_type)value;
+		case 'c':
+			if (!parse_number(sh, &value, state->optarg, NULL,
+			    WIFI_TWT_SETUP_CMD_REQUEST, WIFI_TWT_SETUP_CMD_DEMAND)) {
```

Comment:
Minor nit, but the indentation is wrong here and also for the other added lines.
So here the code should look like
```
			if (!parse_number(sh, &value, state->optarg, NULL,
					  WIFI_TWT_SETUP_CMD_REQUEST,
					  WIFI_TWT_SETUP_CMD_DEMAND)) {
```
Is it possible to adjust your editor to indent the lines automatically?

Reply:
Thanks, all updated as comment